packages/core/src/types.ts

```diff
@@ -1712,19 +1712,13 @@ export type InterpreterFrom<
   any,
   infer TResolvedTypesMeta
 >
-  ? Interpreter<TContext, TStateSchema, TEvent, TTypestate, TResolvedTypesMeta>
-  : T extends (
-      ...args: any[]
-    ) => StateMachine<
-      infer TContext,
-      infer TStateSchema,
-      infer TEvent,
-      infer TTypestate,
-      any,
-      any,
-      infer TResolvedTypesMeta
+  ? Interpreter<
+      TContext,
+      TStateSchema,
+      TEvent,
+      TTypestate,
+      MarkAllImplementationsAsProvided<TResolvedTypesMeta>
```

Comment:
The crucial bit is to use `MarkAllImplementationsAsProvided` here.

It shouldn't be possible to "start"/create a service/actor without providing all implementations first. So this thing should be impossible to create as `interpret` won't allow it:
```ts
Interpreter<TContext, any, TEvent, any, TypegenMetaStillExpectingSomeImplementations>
```
Therefore, it makes sense to just mark those as provided here as wherever you might use `InterpreterFrom<typeof machine>` you should only be able to receive valid interpreters (and that is ones with all implementations marked as provided)